pom.xml

```diff
@@ -1201,6 +1201,21 @@
               </rules>
             </configuration>
           </execution>
+          <execution>
+            <id>no-duplicate-dependencies</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <!-- checks for duplicate dependencies in the "dependencies" and "dependencyManagement" sections. -->
+                <!-- dependencies are considered duplicate if they have the same groupId, artifactId, type, and classifier properties -->
+                <!-- Note: enforcer v3.0.0-M3 does not correctly check "dependencyManagement" section for duplicates:
+                     see https://issues.apache.org/jira/browse/MENFORCER-301 -->
```

Comment:
Can we upgrade the enforcer plugin?

Reply:
i don't see any reason why we cannot upgrade it. the last time the maven enforcer plugin was upgraded was in https://github.com/Alluxio/alluxio/pull/11849

Reply:
upgraded in #17912